pkg/operator/starter.go

```diff
@@ -73,14 +74,41 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, operatorNamespace, targetNamespace)
 	staticResourcesController := staticresourcecontroller.NewStaticResourceController(
 		"CSISnapshotStaticResourceController",
+		assets.ReadFile,
+		[]string{},
+		(&resourceapply.ClientHolder{}).WithKubernetes(kubeClient),
+		operatorClient,
+		controllerConfig.EventRecorder,
+	).WithConditionalResources(
 		assets.ReadFile,
 		[]string{
 			"csi_controller_deployment_pdb.yaml",
 			"webhook_deployment_pdb.yaml",
 		},
-		(&resourceapply.ClientHolder{}).WithKubernetes(kubeClient),
-		operatorClient,
-		controllerConfig.EventRecorder,
+		func() bool {
+			isSNO, precheckSucceeded, err := guard.IsSNOCheckFnc(configInformers.Config().V1().Infrastructures())()
+			if err != nil {
+				klog.Errorf("IsSNOCheckFnc failed: %v", err)
+				return false
+			}
+			if !precheckSucceeded {
+				klog.V(4).Infof("IsSNOCheckFnc precheck did not succeed, skipping")
+				return false
+			}
+			return !isSNO
```

Comment:
Can you create `isSNO()` out of this and use `isSNO()` / `!isSNO` below?

Reply:
Unfortunately, the create/delete conditionals are not opposite of each other. They both can be false when:
- `configInformers.Config().V1().Infrastructures()`'s hasSynced is false
- retrieval of the `cluster` infrastructure object failed.